plugins/network-elements/nuage-vsp/src/com/cloud/network/resource/NuageVspResourceConfiguration.java

```diff
@@ -0,0 +1,310 @@
+                .zoneId((String)configuration.get(ZONE_ID))
+                .hostName((String)configuration.get(HOST_NAME))
+                .cmsUser((String)configuration.get(CMS_USER))
+                .cmsUserPassword((String)configuration.get(CMS_USER_PASSWORD))
+                .port((String)configuration.get(PORT))
+                .apiVersion((String)configuration.get(API_VERSION))
+                .apiRelativePath((String)configuration.get(API_RELATIVE_PATH))
+                .retryCount((String)configuration.get(RETRY_COUNT))
+                .retryInterval((String)configuration.get(RETRY_INTERVAL))
+                .nuageVspCmsId((String)configuration.get(NUAGE_VSP_CMS_ID));
+    }
+
+    private void verifyNotNull(String name, String value) throws ConfigurationException {
+        if (value == null) {
+            throw new ConfigurationException("Unable to find " + name);
+        }
+    }
+
+    private void verifyNotEmpty(String name, String value) throws ConfigurationException {
+        if (org.apache.commons.lang.StringUtils.isEmpty(value)) {
+            throw new ConfigurationException("Unable to find " + name);
+        }
+    }
+
+    public void validate() throws ConfigurationException {
+        verifyNotNull("name", _name);
+        verifyNotNull("guid", _guid);
+        verifyNotNull("zone", _zoneId);
+        verifyNotNull("hostname", _hostName);
+        verifyNotNull("CMS username", _cmsUser);
+        verifyNotNull("CMS password", _cmsUserPassword);
+
+        if (org.apache.commons.lang.StringUtils.isBlank(_apiVersion)) {
+            throw new ConfigurationException("Unable to find API version");
+        }
+
+        try {
+                new NuageVspApiVersion(_apiVersion);
+        } catch(IllegalArgumentException e) {
+            throw new ConfigurationException("Incorrect API version");
+        }
+
+        verifyNotEmpty("number of retries", _retryCount);
+        verifyNotEmpty("retry interval", _retryInterval);
+    }
+
+    public int parseRetryCount() throws ConfigurationException {
+        verifyNotEmpty("number of retries", _retryCount);
+        int retryCount;
+        try {
+            retryCount = Integer.parseInt(_retryCount);
+        } catch (NumberFormatException ex) {
+            throw new ConfigurationException("Number of retries has to be between 1 and 10");
+        }
+        if ((retryCount < 1) || (retryCount > 10)) {
+            throw new ConfigurationException("Number of retries has to be between 1 and 10");
+        }
+        return retryCount;
+
+    }
+
+    public int parseRetryInterval() throws ConfigurationException {
+        verifyNotEmpty("retry interval", _retryInterval);
+        int retryInterval;
+        try {
+            retryInterval = Integer.parseInt(_retryInterval);
+        } catch (NumberFormatException ex) {
+            throw new ConfigurationException("Number of retries has to be between 1 and 10");
```

Comment:
This message appears to be a duplication of line 258 and doesn't match the range on line 275.